x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/action/EsqlCapabilities.java

```diff
@@ -304,7 +304,12 @@ public enum Cap {
         /**
          * Support explicit casting from string literal to DATE_PERIOD or TIME_DURATION.
          */
-        CAST_STRING_LITERAL_TO_TEMPORAL_AMOUNT;
+        CAST_STRING_LITERAL_TO_TEMPORAL_AMOUNT,
+
+        /**
+         * Supported the text categorization function "CATEGORIZE".
+         */
+        CATEGORIZE(true);
 
```

Comment:
Question: does [this piece of code](https://github.com/elastic/elasticsearch/blob/main/x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/CsvTests.java#L262-L269) work as intended?

If you add a non-existing required capability to a CSV test, it always passes.

Reply:
There's a capability presence check here:
https://github.com/elastic/elasticsearch/blob/bb872e658a9f7ad1c154c1c1ef10c1f0b892336b/x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/CsvTests.java#L255-L261
It was moved to snapshot-only for reasons I don't remember. It probably shouldn't. 
I'm not sure if the CI runs a snapshot or not actually (I guess not); it feels weird that it didn't fail in any part of the process

Reply:
I think the CI runs snapshot, and therefore it succeeded

Reply:
There's a tag you can add to make the tests run in release mode if you want.

Reply:
I've run the test locally with and without snapshot and both work, so I think we're good.